### DISCUSSION TITLE
Why is there a desktop app requirement?

### DISCUSSION THREAD

**Discussion by jikkuatwork** [modelcontextprotocol/.github#15] [Ideas]
Can't a simple CLI server & browser plugin achieve this? Making it available to any OS?

**Comment by ezynda3** [comment]
There is no hard requirement the standard is open and available for anyone to implement their own tools. This is still very new but I expect a lot of cool tools to be launched from this soon.

**Comment by jikkuatwork** [comment]
True, but I feel the current implementation target feels less than ideal as it only covers a subset of users; a plugin based system would have covered a much larger segment of users with plugin being able to update much quicker.

**Reply by dsp-ant** [reply]
A plugin system solves a very different set of issues. MCP aims to be an open protocol for any client and any server. The current implementation is just one way to implement this. It is not a Claude.AI specific feature. We aimed to provide an implementation that is reasonably secure, and wide enough available, while allow for flexibility. We understand the desktop requirement currently limits the availability. Please feel free to implement your own client on any platform you wish.

**Reply by jikkuatwork** [reply]
Thanks, I still don't understand the first target implementation was mac/desktop app. Anyways, I will dig deeper and see if I can build a browser plugin client.
